### DISCUSSION TITLE
Is there any option in "infer.mle.fit" that can remove a certain constraint term?

### DISCUSSION THREAD

**Discussion by zwu0922** [scikit-hep/pyhf#1606] [Q&A] [answered]
In Roofit "RooAbsPdf::fitTo()", one would be able to apply constraints only to a given subset of parameters by using the "Constrain(const RooArgSet&pars)" option. I would like to know if there is a counterpart in pyhf.infer.mle.fit?

**Comment by alexander-held** [comment, marked as answer]
This is currently not possible. Here is a related issue: https://github.com/scikit-hep/pyhf/issues/820.

**Reply by zwu0922** [reply]
Thanks a lot, Alexander. Seems that I cannot continue my work with pyhf.

**Reply by lukasheinrich** [reply]
do you want to remove the constraint but keep the NP?

**Reply by zwu0922** [reply]
Yes, I would like to keep the NP for a "histosys" but remove the corresponding constraint term from the likelihood.

**Reply by kratsg** [reply]
Just to briefly chime in, as mentioned in the issue, it's not a technically difficult thing to implement. However, the problem is that you then enter a regime where your model is not necessarily HiFa, but is "HiFa-like". HistFactory is specific about the constraints being incorporated as part of the p.d.f.

I suspect you'll be able to progress more in the corresponding python-RooStats (if/whenever that ends up being fleshed out/developed) rather than particularly in `pyhf` alone. We probably could add support for this, or at least provide an easier way for the user to control or modify the underlying generated HiFa p.d.f. to do similar manipulations.

**Reply by alexander-held** [reply]
It depends a bit on how strictly you define HistFactory, but it is (partially, I think only for OverallSys and not for HistoSys) supported in the xmls (see example in https://github.com/scikit-hep/pyhf/issues/820).
